src/experiments.js

```diff
@@ -226,10 +222,93 @@ Test.registerGoal( 'click', ( element, record ) => {
 	} );
 }, [ 'a' ] );
 
-// Define custom elements.
-window.customElements.define( 'ab-test', ABTest );
+class PersonalizationBlock extends HTMLElement {
+
+	get clientId() {
+		return this.getAttribute( 'client-id' );
+	}
+
+	connectedCallback() {
+		// Set default styles.
+		this.attachShadow( { mode: 'open' } );
+		this.shadowRoot.innerHTML = `
+			<style>
+				:host {
+					display: block;
+				}
+			</style>
+			<slot></slot>
+		`;
+
+		// Update the component content.
+		this.setContent();
+
+		// Attach a listener to update the content when audiences are changed.
+		window.Altis.Analytics.on( 'updateAudiences', this.setContent );
+	}
+
+	setContent = () => {
+		const audiences = window.Altis.Analytics.getAudiences() || [];
+
+		// Track whether we need to use the fallback.
+		let fallback = true;
```

Comment:
Could `fallback` be replaced with `! audience`?

Reply:
No, `! audience` is true if you haven't selected an audience yet. I needed a simple way to differentiate what was the fallback template and what was a true variant without an audience set yet.

Reply:
Oh sorry, was thinking this was a different file, yeah it can be in this case.